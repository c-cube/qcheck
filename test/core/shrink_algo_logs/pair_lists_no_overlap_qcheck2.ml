open QCheck2

let pair_lists_no_overlap =
  Test.make ~name:"pairs lists no overlap"
    Gen.(pair (list small_nat) (list small_nat))
    (Log.shrinks Print.(pair (list int) (list int)) @@
       fun (xs,ys) -> List.for_all (fun x -> not (List.mem x ys)) xs)

let rand = Random.State.make [|1234|];;
try Test.check_exn ~rand pair_lists_no_overlap with Test.Test_fail _ -> ()
