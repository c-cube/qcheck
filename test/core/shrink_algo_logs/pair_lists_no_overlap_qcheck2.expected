fails ([8; 81; 5; 3; 61; 1; 8; 3; 8; 8; 13; 6; 8; 6; 8; 2; 4; 9; 0; 55; 21; 6; 6; 0; 6; 1; 5; 5; 9; 20; 3; 7; 86; 2; 38; 2; 9; 0; 2; 5; 1; 8; 5; 4; 1; 4; 3; 48; 0; 8; 80; 4; 0; 1; 6; 0; 6; 4; 9; 4; 39; 4; 7; 6; 23; 20; 47; 2; 5; 3; 1; 7; 14; 1; 2; 94; 0; 2; 6; 38; 5; 18; 5; 31; 0; 5; 4; 30; 2; 8; 19; 0; 8; 1; 7], [9; 3; 94; 3])
holds ([], [9; 3; 94; 3])
fails ([52; 7; 29; 6; 6; 1; 8; 4; 2; 2; 2; 47; 6; 8; 4; 7; 9; 9; 10; 0; 6; 74; 8; 74; 2; 4; 4; 50; 8; 86; 9; 5; 3; 99; 0; 5; 0; 0; 3; 5; 0; 5; 7; 2; 71; 51; 1], [9; 3; 94; 3])
holds ([], [9; 3; 94; 3])
fails ([6; 4; 1; 91; 4; 60; 91; 4; 3; 84; 4; 7; 9; 1; 9; 7; 3; 2; 7; 57; 76; 8; 9], [9; 3; 94; 3])
holds ([], [9; 3; 94; 3])
fails ([63; 0; 1; 3; 5; 0; 8; 7; 64; 5; 39], [9; 3; 94; 3])
holds ([], [9; 3; 94; 3])
fails ([0; 6; 4; 8; 9], [9; 3; 94; 3])
holds ([], [9; 3; 94; 3])
holds ([1; 80], [9; 3; 94; 3])
holds ([0; 15; 30], [9; 3; 94; 3])
holds ([14; 1; 5; 4], [9; 3; 94; 3])
fails ([0; 0; 4; 8; 9], [9; 3; 94; 3])
fails ([0; 0; 0; 8; 9], [9; 3; 94; 3])
fails ([0; 0; 0; 0; 9], [9; 3; 94; 3])
holds ([0; 0; 0; 0; 0], [9; 3; 94; 3])
holds ([0; 0; 0; 0; 4], [9; 3; 94; 3])
holds ([0; 0; 0; 0; 6], [9; 3; 94; 3])
holds ([0; 0; 0; 0; 7], [9; 3; 94; 3])
holds ([0; 0; 0; 0; 8], [9; 3; 94; 3])
holds ([0; 0; 0; 0; 9], [])
holds ([0; 0; 0; 0; 9], [5; 51])
holds ([0; 0; 0; 0; 9], [7; 87; 7])
fails ([0; 0; 0; 0; 9], [0; 3; 94; 3])
fails ([0; 0; 0; 0; 0], [0; 3; 94; 3])
fails ([0; 0; 0; 0; 0], [0; 0; 94; 3])
fails ([0; 0; 0; 0; 0], [0; 0; 0; 3])
fails ([0; 0; 0; 0; 0], [0; 0; 0; 0])
